### PR TITLE
RDoc-1770 - document attachments multi-get support

### PR DESCRIPTION
**Pull request by reebhub**

### CODE REVIEW

File: Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs [hunk 1]
```diff
@@ -212,6 +220,142 @@ public class Album
...
+                        string userId = result[userCnt].Id;
+                        #region GetAllAttachments
+                        // Load a user profile
+                        var user = session.Load<User>(userId);
+
+                        // Get the names of files attached to thie document
+                        IEnumerable<AttachmentRequest> attachmentNames = session.Advanced.Attachments.GetNames(user).Select(x => new AttachmentRequest(userId, x.Name));
+
+                        // Get the attached files
+                        IEnumerator<AttachmentEnumeratorResult> attachmentsEnumerator = session.Advanced.Attachments.Get(attachmentNames);
```
Comment (ppekrol) [inline Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs:303]: We need to expand this example. Show how we can enumerate and what is inside the 'AttachmentEnumeratorResult'

File: Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs [hunk 2]
```diff
@@ -212,6 +220,172 @@ public class Album
...
+                using (var session = store.OpenSession())
+                {
+                    for (var userCnt = 0; userCnt < result.Count; userCnt++)
+                    {
+                        string userId = result[userCnt].Id;
+                        #region GetAllAttachments
+                        // Load a user profile
+                        var user = session.Load<User>(userId);
+
+                        // Get the names of files attached to thie document
```
Comment (garayx) [inline Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs:299]:
```suggestion
                        // Get the names of files attached to this document

```

File: Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs [hunk 3]
```diff
@@ -212,6 +220,172 @@ public class Album
...
+                using (var session = store.OpenAsyncSession())
+                {
+                    for (var userCnt = 0; userCnt < result.Count; userCnt++)
+                    {
+                        string userId = result[userCnt].Id;
+                        #region GetAllAttachmentsAsync
+                        // Load a user profile
+                        var user = await session.LoadAsync<User>(userId);
+
+                        // Get the names of files attached to thie document
```
Comment (garayx) [inline Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs:333]:
```suggestion
                        // Get the names of files attached to this document

```

**Review by garayx — APPROVED** [review]

File: Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs (hunk 1 above)
Comment (reebhub) [reply Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs:303]: done

File: Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs (hunk 2 above)
Comment (reebhub) [reply Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs]: fixed

File: Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs (hunk 3 above)
Comment (reebhub) [reply Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Attachments/Attachments.cs]: fixed

**Pull request merged**